src/codeflare_sdk/utils/generate_yaml.py

```diff
@@ -51,35 +51,200 @@ def gen_names(name):
+            except Exception as e:  # pragma: no cover
+                return _kube_api_error_handling(e)
+            domain = ingress["spec"]["domain"]
+        elif ingress_domain is None:
+            raise ValueError(
+                "ingress_domain is invalid. For Kubernetes Clusters please specify an ingress domain"
+            )
+        else:
+            domain = ingress_domain
+        del metadata["annotations"]
+        spec["rules"][0]["host"] = f"ray-dashboard-{cluster_name}-{namespace}.{domain}"
+
+
+def update_rayclient_ingress(
+    ingress_item, cluster_name, namespace, ingress_options, ingress_domain
+):  # pragma: no cover
+    metadata = ingress_item.get("generictemplate", {}).get("metadata")
+    spec = ingress_item.get("generictemplate", {}).get("spec")
+    if ingress_options != {}:
+        for index, ingress_option in enumerate(ingress_options["ingresses"]):
+            if "ingressName" not in ingress_option.keys():
+                raise ValueError(
+                    f"Error: 'ingressName' is missing or empty for ingress item at index {index}"
+                )
+            if "port" not in ingress_option.keys():
+                raise ValueError(
+                    f"Error: 'port' is missing or empty for ingress item at index {index}"
+                )
+            elif not isinstance(ingress_option["port"], int):
+                raise ValueError(
+                    f"Error: 'port' is not of type int for ingress item at index {index}"
+                )
+            if ingress_option["port"] == 10001:
+                metadata["name"] = ingress_option["ingressName"]
+                metadata["namespace"] = namespace
+                if "annotations" not in ingress_option.keys():
+                    del metadata["annotations"]
+                else:
+                    metadata["annotations"] = ingress_option["annotations"]
+                if "path" not in ingress_option.keys():
+                    del spec["rules"][0]["http"]["paths"][0]["path"]
+                else:
+                    spec["rules"][0]["http"]["paths"][0]["path"] = ingress_option[
+                        "path"
+                    ]
+                if "pathType" not in ingress_option.keys():
+                    spec["rules"][0]["http"]["paths"][0][
+                        "pathType"
+                    ] = "ImplementationSpecific"
+                if "host" not in ingress_option.keys():
+                    del spec["rules"][0]["host"]
+                else:
+                    spec["rules"][0]["host"] = ingress_option["host"]
+                if "ingressClassName" not in ingress_option.keys():
+                    del spec["ingressClassName"]
+                else:
+                    spec["ingressClassName"] = ingress_option["ingressClassName"]
+
+                spec["rules"][0]["http"]["paths"][0]["backend"]["service"][
+                    "name"
+                ] = f"{cluster_name}-head-svc"
```

Comment:
@Bobbins228  The interactive notebook seems to work on KinD with my specified ingress_domain and OCP with the default ingress_domain. 

One question is do we have a need to customize the rayclient Ingress using the ingress_options? I am almost certain I would not be able to connect to this customized Ingress because the required annotations and [alt_name] DNS.x not containing the host in the head initContainer command.

Reply:
I can remove the ability to customise the client ingress I don't necessarily see the need for a custom client ingress either. The only changes needed are for if a user is on kind or OpenShift with regards to annotations and class names but we can do that ourselves.

Reply:
I am not familiar with the full functionality of the annotations and class. You may make the call.